src/hooks/user.hook.ts

```diff
@@ -0,0 +1,61 @@
+import {useMutation, gql} from '@apollo/client';
+import {useNavigation} from '@react-navigation/native';
+import {NavigationProp} from '@react-navigation/native';
+import {Alert} from 'react-native';
+import {RootStackParamList} from '../types/navigation';
+
+const CREATE_USER_MUTATION = gql`
+  mutation CreateUser($data: UserInput!) {
+    createUser(data: $data) {
+      id
+      name
+      email
+      phone
+      birthDate
+      role
+    }
+  }
+`;
+
+interface User {
+  name: string;
+  email: string;
+  phone: string;
+  birthDate: string;
+  role: string;
+  password: string;
+}
+
+export const useCreateUser = () => {
+  const [createUserMutation] = useMutation(CREATE_USER_MUTATION);
+  const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+
+  const createUser = async (user: User) => {
+    try {
+      const {birthDate, ...rest} = user;
```

Comment:
```suggestion
      const {birthDate, ...userWithoutBirthDate} = user;
```

Pode parecer um pouco redundante, mas acho que ajuda a dar clareza na leitura.